Automattic/wp-calypso
# Plans (State): Remove unused props from Redux SitePlan object/assembler

Part of addressing https://github.com/Automattic/wp-calypso/issues/86638

## Proposed Changes

Clean up `SitePlanData` object by removing unused properties to help with migration to data-stores. Part of https://github.com/Automattic/wp-calypso/issues/86638 may later involve updating endpoint code to not send these values through, although this is strictly for frontend references.

The following props appear unused and are being removed from the object and assembler:

- `canStartTrial`
- `discountReason`
- `formattedDiscount`
- `formattedOriginalPrice`
- `formattedPrice`
- `isDomainUpgrade`

## Why are these changes being made?


We only need one form of handling plan pricing in the code base, more so one framework. Anything else risks having alternative and differing views of the same data propagating across the code base.

## Testing Instructions



* Build passing
* Confirm neither the property names nor the API snake_cased variables are referenced anywhere in code (untyped files or optional operators may go unnoticed in build)

## Pre-merge Checklist
